sklearn/feature_selection/tests/test_variance_threshold.py

```diff
@@ -33,6 +34,13 @@ def test_variance_threshold():
         assert (len(data), 1) == X.shape
 
 
+def test_variance_negative():
+    # Test VarianceThreshold with negative variance.
+    for X in [data, csr_matrix(data)]:
+        with pytest.raises(ValueError):
+            VarianceThreshold(threshold=-1.).fit(X)
```

Comment:
Mostly style changes to move to pytest:

```suggestion
@pytest.mark.parametrize(X, [data, csr_matrix(data)])
def test_variance_negative(X):
    """Test VarianceThreshold with negative variance."""
    var_threshold = VarianceThreshold(threshold=-1.)
    msg = "Threshold must be non-negative"
    with pytest.raises(ValueError, match=msg):
        var_threshold.fit(X)
```

Also this change explicitly makes sure the error is raised when `fit` is called.

Reply:
Thank you. I changed as above and slightly modified `msg` to include a value used when an error occurs.

And also I added regex signals to make sure that the test doesn't pass when, for example, the message is `Threshold must be non-negative. Got: -1.0....`. 

`msg` without regex signals would return match when a string with arbitrary characters before or after the pattern was passed.

But, I feel this fix may be redundant. What do you think?